docs/setup-guide.md

```diff
@@ -0,0 +1,371 @@
+#!/bin/sh
+
+echo -e “Hello Evil!”
+EOF
+
+[root@keylime-fedora ~]# chmod +x bad-script.sh
+[root@keylime-fedora ~]# ./bad-script.sh
+```
+
+## Revocation
+In this section, you will set up a custom revocation action on the agent by including a revocation script as part of the payload that is sent from the tenant to the agent. This payload can be decrypted and executed by the agent upon its first successful attestation. After this first successful attestation, if the agent fails an attestation, the revocation actions will be executed.
+
+### Set Up Payload
+
+First, make sure that the `payload_script` in the `[cloud_agent]` section of `/etc/keylime.conf` is set to `autorun.sh`:
+
+```
+[cloud_agent]
+
+payload_script=autorun.sh
+```
+
+This specifies the script to run on the agent after its first successful attestation.
+
+#### Option 1: Quick and easy payload
+
+Note this example uses the default values for the tenant command, but you may substitute your own:
+
+```shell
+[root@keylime-fedora ~]# mkdir /root/payload
+[root@keylime-fedora ~]# echo "echo -e 'hello'" > /root/payload/autorun.sh
+
+[root@keylime-fedora ~]# keylime_tenant -v 127.0.0.1 -t 127.0.0.1 --uuid d432fbb3-d2f1-4a97-9ef7-75bd81c00000 --allowlist /root/allowlist.txt --include /root/payload --cert /root/ca --exclude /root/excludes.txt -c add
+```
+
+#### Option 2: More realistic payload
+
+For this payload, you will need to use or [generate a key pair](https://www.ssh.com/academy/ssh/keygen) for ssh (the private key will be called `id_rsa` and the public key will be called `id_rsa.pub`).
+
+```shell
+[root@keylime-fedora ~]# mkdir /root/payload && cd /root/payload
+
+# set up action_list
+[root@keylime-fedora ~]# echo “local_action_remove_ssh” > action_list
+
+# set up autorun.sh
+[root@keylime-fedora ~]# cat > autorun.sh <<EOF
+> #!/bin/sh
+> mkdir -p /root/.ssh/
+> cp id_rsa* /root/.ssh/
+> chmod 600 /root/.ssh/id_rsa*
+> EOF
+
+
+# set up local_action_remove_ssh.py
+[root@keylime-fedora ~]# cat > local_action_remove_ssh.py <<EOF
+> import os
+> import asyncio
+>
+> async def execute(revocation):
+>	if revocation['type']!='revocation':
+>    	return
+>	os.remove("/root/.ssh/id_rsa")
+>	os.remove("/root/.ssh/id_rsa.pub")
+> EOF
+```
+
+At this point, the payload directory should look like this:
+
+```shell
+[root@keylime-fedora ~]# ls /root/payload
+
+action_list autorun.sh id_rsa id_rsa.pub local_action_remove_ssh.py
+```
+
+Now execute the tenant command including the CA:
+
+```shell
+[root@keylime-fedora ~]# keylime_tenant -v 127.0.0.1 -t 127.0.0.1 -u d432fbb3-d2f1-4a97-9ef7-75bd81c00000 --cert /root/myca --allowlist /root/allowlist.txt --exclude /root/excludes.txt -f—-include /root/payload -c add
+```
+
+Run keylime_ca
+
+```shell
+[root@keylime-fedora ~]# keylime_ca -c listen -d /root/myca
+```
+
+## Troubleshooting
+
+Here are some solutions to possible errors bringing the Vagrant box up encountered on a clean Fedora 36 install.
+
+```shell
```

Comment:
Since this is a list of possible errors that user may encounter, there needs to be clear delineation between each possible error otherwise it's easy to accidentally continue with steps from an unrelated error. Maybe use a numbered header for each? Or give them a sub-title of some kind?

Reply:
Done in my latest commit, and I did the same for the multi-VM setup section.